src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1268,6 +1273,51 @@ static void resume_pending_batch_in_call_combiner(void* arg,
   grpc_subchannel_call_process_op(subchannel_call, batch);
 }
 
+// The callback to intercept trailing metadata if retries is not enabled
+static void recv_trailing_metadata_ready_for_lb(void* arg, grpc_error* error) {
+  subchannel_batch_data* batch_data = static_cast<subchannel_batch_data*>(arg);
+  grpc_call_element* elem = batch_data->elem;
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  GPR_ASSERT(calld->pick.recv_trailing_metadata_ready != nullptr);
+  GPR_ASSERT(calld->pick.recv_trailing_metadata != nullptr);
+
+  GRPC_CLOSURE_SCHED(
+      calld->pick.recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+  calld->pick.recv_trailing_metadata = nullptr;
+  calld->pick.recv_trailing_metadata_ready = nullptr;
+
+  GRPC_CLOSURE_RUN(
+      calld->before_lb_intercept_recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+}
+
+// Installs a interceptor to inform the lb of the trailing metadata, if needed
+static void maybe_intercept_trailing_metadata_for_lb(
+    void* arg, grpc_transport_stream_op_batch* batch) {
+  subchannel_batch_data* batch_data = static_cast<subchannel_batch_data*>(arg);
+  grpc_call_element* elem = batch_data->elem;
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  if (calld->pick.recv_trailing_metadata_ready != nullptr) {
+    GPR_ASSERT(calld->pick.recv_trailing_metadata != nullptr);
+    // Unlike the retries case, the location of the trailing metadata is known
```

Comment:
It might be better to do this in the callback, just for consistency between the two code paths.  That way, if in the future we change the filter API such that the batch can get changed by filters beneath us, we won't need to change this later.